pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -697,6 +731,10 @@ func handleDomainDelete(ctxArg interface{}, key string,
 	configArg interface{}) {
 
 	log.Infof("handleDomainDelete(%s)\n", key)
+	// delete the specific cipher block status
+	ctx := ctxArg.(*domainContext)
+	config := configArg.(types.DomainConfig)
+	unpublishCipherBlockStatus(ctx, config.Key())
```

Comment:
Note that you only publish if you had cloudinit user data, but you unconditionally unpublish. Hence you get some log.Error noise which is undesriable

Reply:
Have put a check, to not to publish, incase no valid cloud init user data (cipher or, plain)